url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/DeviceGraphics/KitRobotGraphics.java

```diff
@@ -0,0 +1,269 @@
+							tempKitGraphics);
+					server.sendData(new Request(
+							Constants.STAND_GIVE_KIT_COMMAND,
+							Constants.STAND_TARGET + 1, null));
+				} else {
+					tempKitGraphics.setPosition(0);
+					kitPositions.put(Constants.KIT_INSPECTION_AREA,
+							tempKitGraphics);
+					server.sendData(new Request(
+							Constants.STAND_GIVE_KIT_COMMAND,
+							Constants.STAND_TARGET + 2, null));
+				}
+			}
+
+		}
+
+		kitPositions.put(Constants.KIT_INSPECTION_AREA, kit);
+
+	}
+
+	public void msgPlaceKitOnStand1(KitGraphics kit) {
+		kit.setLocation(location1);
+		kit.setPosition(1);
+		kitPositions.put(Constants.KIT_LOCATION1, kit);
+		server.sendData(new Request(
+				Constants.KIT_ROBOT_DISPLAY_PICKS_CONVEYOR_TO_LOCATION1,
+				Constants.KIT_ROBOT_TARGET, null));
+
+	}
+
+	public void msgPlaceKitOnStand2(KitGraphics kit) {
+		kit.setLocation(location2);
+		kit.setPosition(2);
+		kitPositions.put(Constants.KIT_LOCATION2, kit);
+		server.sendData(new Request(
+				Constants.KIT_ROBOT_DISPLAY_PICKS_CONVEYOR_TO_LOCATION2,
+				Constants.KIT_ROBOT_TARGET, null));
+	}
+
+	/*
+	 * messages KitRobotGraphicsDisplay to move a kit to the GoodConveyor
+	 * (non-Javadoc)
+	 * 
+	 * @see GraphicsInterfaces.KitRobotGraphics#msgPlaceKitOnConveyor()
+	 */
+	@Override
+	public void msgPlaceKitOnConveyor() {
+		kitPositions.get(Constants.KIT_INSPECTION_AREA).setPosition(3);
+		kitPositions.put(Constants.KIT_INSPECTION_AREA, null);
+		server.sendData(new Request(Constants.STAND_GIVE_KIT_COMMAND,
+				Constants.STAND_TARGET + 0, null));
+	}
+
+	/*
+	 * Moves the kit to a stand location (non-Javadoc)
+	 * 
+	 * @see
+	 * GraphicsInterfaces.KitRobotGraphics#msgPlaceKitOnStand(DeviceGraphics
+	 * .KitGraphics, int)
+	 */
+	@Override
+	public void msgPlaceKitOnStand(KitGraphics kit, int location) {
+		Boolean placeFromInspection = false;
+		if(kitPositions.get(Constants.KIT_INSPECTION_AREA)==kit)
+		{
+			placeFromInspection=true;	
+		}
+		
+		System.out.println("dfjklqhwetjhqweiponqwerihadfjkghuiafghuioqhejtoiqwewqehtuijklllllllllllllllllllllllllllllllllllllllllqwetiohqweuithqweuiothj");
+		if ( !placeFromInspection ){
+			if ( location == 1) {
+				msgPlaceKitOnStand1(kit);
+			} else if ( location == 2) {
+				msgPlaceKitOnStand2(kit);
+			}
+		} else {
+			if ( location == 1 ){
+				kit.setPosition(1);
+				kitPositions.put(Constants.KIT_LOCATION1, kit);
+			} else if( location == 2){
+				kit.setPosition(2);
+				kitPositions.put(Constants.KIT_LOCATION2, kit);
+			}
+			
+			server.sendData(new Request(Constants.STAND_GIVES_BACK_TO_ANOTHER_STAND, Constants.STAND_TARGET + 0 , location) );	
+			kitPositions.put(Constants.KIT_INSPECTION_AREA, null);
+		}
+		
+
+	}
+
+	/*
+	 * receives request that will send commands based on the request
+	 * 
+	 * @see DeviceGraphics.DeviceGraphics#receiveData(Networking.Request)
+	 * request is a class that has a command target and data
+	 */
+	@Override
+	public void receiveData(Request req) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.